skl2onnx/algebra/type_helper.py

```diff
@@ -31,6 +31,9 @@ def _guess_dim(value):
 
     if isinstance(given_type, (np.ndarray, coo_matrix)):
         shape = list(given_type.shape)
+        if len(shape) == 0:
```

Comment:
if not shape: ?

Reply:
I usually avoid not shape or shape as I know shape is not None. That means testing one less condition.

Reply:
pylint would complain about it :)

Reply:
flake8 is used instead of pylint. We can switch but maybe in another PR as I expect many lint issues to pop up.

Reply:
no worry, I'm nit picking.